modules/publicize/publicize.php

```diff
@@ -510,6 +510,210 @@ function test_publicize_conns() {
+	 *
+	 * @return array {
+	 *     Array of UI setup data for connection list form.
+	 *
+	 *     @type string 'unique_id'     ID string representing connection
+	 *     @type string 'service_name'  Slug of the connection's service (facebook, twitter, ...)
+	 *     @type string 'service_label' Service Label (Facebook, Twitter, ...)
+	 *     @type string 'display_name'  Connection's human-readable Username: "@jetpack"
+	 *     @type bool   'enabled'       Default value for the connection (e.g., for a checkbox).
+	 *     @type bool   'done'          Has this connection already been publicized to?
+	 *     @type bool   'toggleable'    Is the user allowed to change the value for the connection?
+	 *     @type bool   'global'        Is this connection a global one?
+	 * }
+	 */
+	public function get_filtered_connection_data( $selected_post_id = null ) {
+		$connection_list = array();
+
+		$post = get_post( $selected_post_id ); // Defaults to current post if $post_id is null.
+		// Handle case where there is no current post.
+		if ( ! empty( $post ) ) {
+			$post_id = $post->ID;
+		} else {
+			$post_id = null;
+		}
+
+		$services = $this->get_services( 'connected' );
+		$all_done = $this->done_sharing_post( $post_id );
+
+		// We don't allow Publicizing to the same external id twice, to prevent spam.
+		$service_id_done = (array) get_post_meta( $post_id, $this->POST_SERVICE_DONE, true );
+
+		foreach ( $services as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $this->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+
+				$unique_id = $this->get_connection_unique_id( $connection );
+
+
+				// Was this connection (OR, old-format service) already Publicized to?
+				$done = ! empty( $post ) && (
+					// New flags
+					1 == get_post_meta( $post->ID, $this->POST_DONE . $unique_id, true )
+					||
+					// old flags
+					1 == get_post_meta( $post->ID, $this->POST_DONE . $service_name, true )
+				);
+
+				/**
+				 * Filter whether a post should be publicized to a given service.
+				 *
+				 * @module publicize
+				 *
+				 * @since 2.0.0
+				 *
+				 * @param bool true Should the post be publicized to a given service? Default to true.
+				 * @param int $post_id Post ID.
+				 * @param string $service_name Service name.
+				 * @param array $connection_data Array of information about all Publicize details for the site.
+				 */
+				if ( ! apply_filters( 'wpas_submit_post?', true, $post_id, $service_name, $connection_data ) ) {
+					continue;
+				}
+
+				// Should we be skipping this one?
+				$skip = (
+					(
+						! empty( $post )
+						&&
+						in_array( $post->post_status, array( 'publish', 'draft', 'future' ) )
+						&&
+						(
+							// New flags
+							get_post_meta( $post_id, $this->POST_SKIP . $unique_id, true )
+							||
+							// Old flags
+							get_post_meta( $post->ID, $this->POST_SKIP . $service_name )
```

Comment:
See https://github.com/Automattic/jetpack/pull/10396/files#r228333272 -- here it's even a bit stranger since [R598](https://github.com/Automattic/jetpack/pull/10396/files#diff-2f5f58835a08193f26adccabf4ae5ef3R598) is actually using `$post_id` already...